### PR TITLE
E2e backstage test case

### PR DESCRIPTION
**Pull request by cmoulliard**
## Introduction

- E2e backstage test case
- #81 

## Job description

The github workflow currently do this

1. Setup a kube cluster with ArgoCD, Tekton, KubeVirt, Gitea, Ingress and configure the components
2. Install the DataVolume of the fedora podman image and create a Virtual Machine running podman
3. Git clone backstage-playground
4. Install Node and build backstage-playground
5. Launch backstage using `yarn dev`
6. Scaffold a quarkus project using as template: Quarkus application with the data provided under the folder test-data
7. Wait till the backstage component is created
8. Wait till all we got all the events for the taskId created by backstage
9. Wait till ArgoCD resources are sync
10. Wait till Tekton pipeline succeeded
11. Check about the Quarkus Application deployed and log it

### CODE REVIEW

**Comment by iocanel** [comment]
@cmoulliard: is this ready for review?

File: locations/templates/quarkus-application/template.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ spec:
           title: Name
           type: string
           description: Unique name of the application
-          default: my-quarkus-app
+          default: my-quarkus-app-job
```
Comment (iocanel) [inline locations/templates/quarkus-application/template.yaml]: why?

File: locations/templates/quarkus-application/template.yaml [hunk 2]
```diff
@@ -175,17 +175,32 @@ spec:
             host:
               title: Repo Host
               type: string
-              default: github.com
+              default: gitea.cnoe.localtest.me:8443
```
Comment (iocanel) [inline locations/templates/quarkus-application/template.yaml]: I don't like changing our templates in order to better serve our integration tests. 
It should be the other way around.

File: locations/templates/quarkus-application/template.yaml [hunk 3]
```diff
@@ -175,17 +175,32 @@ spec:
...
+              enum:
+                - gitea.cnoe.localtest.me:8443
```
Comment (iocanel) [inline locations/templates/quarkus-application/template.yaml]: Same as above.

File: locations/templates/quarkus-application/template.yaml [hunk 4]
```diff
@@ -196,24 +211,26 @@ spec:
...
           type: string
           description: The openshift/kubernetes namespace for deploying resources
-          default: iocanel
+          default: dummy
 
         imageRepository:
            title: Select a registry
            type: string
-           default: quay.io
+           default: gitea.cnoe.localtest.me:8443/giteadmin
```
Comment (iocanel) [inline locations/templates/quarkus-application/template.yaml]: Same

File: locations/templates/quarkus-application/template.yaml (hunk 1 above)
Comment (iocanel) [reply locations/templates/quarkus-application/template.yaml]: This is not a `job`, so the suffix here is missleading.

**Review by iocanel — CHANGES_REQUESTED** [review]
This is definitely a step in the right direction.
However, I don't like the fact that the templates have been modified with integration test specific information (e.g. URL to gitea etc). I think that we could just pass those values via `curl` without changing our enums for the UI components, or our defaults values for our UI components.

In the same spirit I don't really like the `dummy` namespace the pull request introduces, but I think that this is secondary.

**Comment by cmoulliard** [comment]
> is this ready for review?

Not yet as the github workflow fails due to a DNS resolution issue using giteas server running on the kind cluster when we create the ArgoCD Applications or when Tekton job runs. I'm discussing this issue with cnoe idpbuilder folks

File: locations/templates/quarkus-application/template.yaml (hunk 1 above)
Comment (cmoulliard) [reply locations/templates/quarkus-application/template.yaml]: When the job will work as github workflow, I will clean the template of this PR. Please wait

File: locations/templates/quarkus-application/template.yaml (hunk 2 above)
Comment (cmoulliard) [reply locations/templates/quarkus-application/template.yaml]: When the job will work as github workflow, I will clean the template of this PR. Please wait

File: locations/templates/quarkus-application/template.yaml (hunk 3 above)
Comment (cmoulliard) [reply locations/templates/quarkus-application/template.yaml]: When the job will work as github workflow, I will clean the template of this PR. Please wait

File: locations/templates/quarkus-application/template.yaml (hunk 4 above)
Comment (cmoulliard) [reply locations/templates/quarkus-application/template.yaml]: When the job will work as github workflow, I will clean the template of this PR. Please wait

**Comment by cmoulliard** [comment]
Can you please review this PR in order to merge i and to check what the action will do as until now we are still blocked by this error ? @iocanel 
```
2024-06-20T14:01:53.3633005Z     "taskId": "4c9974fb-b87f-4ffa-84c6-90f1962ab6ac",
2024-06-20T14:01:53.3633091Z     "body": {
2024-06-20T14:01:53.3639475Z       "message": "InputError: No token available for host: github.com, with owner q-shift, and repo my-quarkus-app-job
at getOctokitOptions (/home/runner/work/backstage-playground/backstage-playground/node_modules/@backstage/plugin-scaffolder-backend-module-github/src/actions/helpers.ts:95:11)
...
2024-06-20T14:01:53.3639800Z       "stepId": "publish-github",
2024-06-20T14:01:53.3639897Z       "status": "failed"
```

**Review by aureamunoz — COMMENTED** [review]
I have the same comment like @iocanel : +1 to pass integration tests values specifically from the command line in the workflows but in general looks good to me.
Also, a high level overview of the changes in the description of the PR would be nice. And squashing the commits the cherry on the cake ;-)

**Comment by cmoulliard** [comment]
> Also, a high level overview of the changes in the description of the PR would be nice

I included, partof the description of the PR, a section detailing what the github workflow job do. Do you want more information ? @aureamunoz

File: locations/templates/quarkus-application/template.yaml (hunk 1 above)
Comment (cmoulliard) [reply locations/templates/quarkus-application/template.yaml]: Name reverted to `default: my-quarkus-app`

File: locations/templates/quarkus-application/template.yaml (hunk 2 above)
Comment (cmoulliard) [reply locations/templates/quarkus-application/template.yaml]: git provider default reverted to `default: github.com`

File: locations/templates/quarkus-application/template.yaml (hunk 3 above)
Comment (cmoulliard) [reply locations/templates/quarkus-application/template.yaml]: First enum value is now
```
              enum:
                - github.com
                ...
```

File: locations/templates/quarkus-application/template.yaml (hunk 4 above)
Comment (cmoulliard) [reply locations/templates/quarkus-application/template.yaml]: ImageRepository reverted too to Quay.io
```
        imageRepository:
           title: Select a registry
           type: string
           default: quay.io
           enum:
             - quay.io
             - gitea.cnoe.localtest.me:8443/giteadmin
             - image-registry.openshift-image-registry.svc:5000
           enumNames:
             - Quay
             - Gitea
             - Internal OpenShift Registry
```

File: locations/templates/quarkus-application/template.yaml [hunk 5]
```diff
@@ -187,7 +187,7 @@ spec:
               default: q-shift
               enum:
                 - q-shift
-                - aureamunoz
+                - amunozhe
```
Comment (aureamunoz) [inline locations/templates/quarkus-application/template.yaml]: For git, it should be `aureamunoz`. `amunozhe` is for quay

**Review by aureamunoz — CHANGES_REQUESTED** [review]

**Pull request merged**